File: CS/App1/UserRole.cs
﻿using Windows.UI.Xaml.Markup;

namespace App1
{
    public enum UserRole
    {
        Administrator,
        Moderator,
        User,
        Director,
        Manager,
        Developer,
        Linquist,
        ProductManager
    }

    public class UserRoleExtension: MarkupExtension
    {
        protected override object ProvideValue()
        {
            return new UserRole();
        }
    }
}

File: VB/App2/UserRole.vb
Imports Windows.UI.Xaml.Markup

Public Enum UserRole
    Administrator
    Moderator
    User
    Director
    Manager
    Developer
    Linguist
    ProductManager
End Enum

Public Class UserRoleExtension
    Inherits MarkupExtension

    Protected Overrides Function ProvideValue() As Object
        Return New UserRole()
    End Function
End Class

